main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java

```diff
@@ -85,23 +86,175 @@ private <E extends GtfsEntity> int loadUniqueCount(
   }
 
   private void loadSpecFeatures(GtfsFeedContainer feedContainer) {
+    loadSpecFeaturesBasedOnFilePresence(feedContainer);
+    loadSpecFeaturesBasedOnFieldPresence(feedContainer);
+  }
+
+  private void loadSpecFeaturesBasedOnFilePresence(GtfsFeedContainer feedContainer) {
+    List<Pair<String, String>> componentList =
+        List.of(
+            new Pair<>("Pathways", GtfsPathway.FILENAME),
+            new Pair<>("Transfers", GtfsTransfer.FILENAME),
+            new Pair<>("Fares V1", GtfsFareAttribute.FILENAME),
+            new Pair<>("Fares V2", GtfsFareProduct.FILENAME),
+            new Pair<>("Shapes", GtfsShape.FILENAME),
+            new Pair<>("Frequency-Based Trip", GtfsFrequency.FILENAME),
+            new Pair<>("Feed Information", GtfsFeedInfo.FILENAME),
+            new Pair<>("Attributions", GtfsAttribution.FILENAME),
+            new Pair<>("Translations", GtfsTranslation.FILENAME),
+            new Pair<>("Fare Media", GtfsFareMedia.FILENAME),
+            new Pair<>("Zone-Based Fares", GtfsStopArea.FILENAME),
+            new Pair<>("Transfers", GtfsTransfer.FILENAME));
+
+    for (Pair<String, String> entry : componentList) {
+      specFeatures.put(
+          entry.getKey(),
+          hasAtLeastOneRecordInFile(feedContainer, entry.getValue()) ? "Yes" : "No");
```

Comment:
The _"Yes"_ and  _"No"_ values can be converted to constants.

Reply:
I updated the code and removed it completely. True & false was good enough for our use if the values.